src/state/events.hpp

```diff
@@ -0,0 +1,96 @@
+#pragma once
+
+namespace genie {
+
+enum Sound_t {
+  SOUND_NO_MATCH = -1,
+  SOUND_MATCH = 0,
+  SOUND_NEWS_INTRO = 1,
+  SOUND_ALARM_CLOCK_ELAPSED = 2,
+};
+
+struct AudioFrame {
+  int16_t *samples;
+  size_t length;
+
+  AudioFrame(size_t len) : samples(new int16_t[len]), length(len) {}
+  ~AudioFrame() { delete[] samples; }
+
+  AudioFrame(const AudioFrame &) = delete;
+  AudioFrame &operator=(const AudioFrame &) = delete;
+
+  AudioFrame(AudioFrame &&other)
+      : samples(other.samples), length(other.length) {
+    other.samples = nullptr;
+    other.length = 0;
+  }
+};
+
+namespace state {
+namespace events {
+
+struct Event {};
+
+struct Wake : Event {};
+
+struct InputFrame : Event {
+  AudioFrame *frame;
```

Comment:
This should probably be an embedded struct rather than a pointer. It would be more idiomatic C++ and safer for memory management
(Moving AudioFrames is cheap because the move constructor just moves the pointer around)

Reply:
Ok I need to learn from you or somewhere else about that stuff. I either haven't ever done move semantics in C++ or it's just been way too long. None of the structs or classes in this PR have move constructors defined, for instance

Reply:
Addressed in cc86f8646790a149bde3512c8b58e2c61fce9123